src/core/or/connection_edge.c

```diff
@@ -1764,6 +1771,46 @@ parse_extended_hostname(char *address, hostname_type_t *type_out)
   return false;
 }
 
+bool lookup_anon_dns_mapping(const char *anon_address, char *onion_address_out) {
+  char *file_content = NULL;
+
+  // Check if the file exists using `file_status`
+  char *dns_fname = get_datadir_fname("anons");
```

Comment:
Might be useful to sanity check the output

Reply:
I did not get it. This is the existing function for data-dir file management

Reply:
I see the file_status is just mapping results. Nvm.